Fix TextEdit::gutters_width Incorrect Value

Tim-Fronsee opened the pull request:
### Summary

#### Bug

When adding & removing a gutter to a `TextEdit`, the total gutter width returned by `TextEdit::get_total_gutter_width` (GDScript) returns the incorrect value.

#### Platform

Windows 10

#### Godot Version

4.x [master](https://github.com/godotengine/godot/commit/6e390fa9abf2101f8772c4ef6694471153105636)

###### Before Add

![image](https://user-images.githubusercontent.com/25569850/186882589-407cad57-717b-4179-9514-8b535bbe8768.png)

###### After Add

![image](https://user-images.githubusercontent.com/25569850/186882681-58135836-191e-4787-b493-7437f414adde.png)

#### Fix

When adding & removing a gutter, `TextEdit::_update_gutter_width` is now called to ensure `TextEdit::gutters_width` has the correct value.

###### After Add

![image](https://user-images.githubusercontent.com/25569850/186881786-1e5ac7b7-4175-41ca-bf42-f4465355590d.png)

###### After Remove

![image](https://user-images.githubusercontent.com/25569850/186882006-ee7b9f83-69d2-48d7-87cc-af8556708711.png)

<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.
-->

Tim-Fronsee committed:
Fix TextEdit::gutters_width (total gutter width) when adding & removing a gutter by calling TextEdit::_update_gutter_width in TextEdit::add_gutter & TextEdit::remove_gutter

Update TextEdit gutters subcase, gutter add and remove to ensure gutter total width is correct

Fix test_code_edit symbol lookup test case to include padding (+2)

Paulb23 reviewed (changes requested):
Nice spot, we shouldn't need to call on `add_gutter` as the default width is `0`, will also need to adjust the unit test to catch this.

Tim-Fronsee commented:
> Nice spot, we shouldn't need to call on `TextEdit::add_gutter` as the default width is `0`, will also need to adjust the unit test to catch this.

When I remove `_update_gutter_width` from `TextEdit::add_gutter`, the gutter width is still not updated correctly.  

This is because `GutterInfo::draw` is `true` by default.  Thus, in `TextEdit::set_gutter_draw` `_update_gutter_width` is not called (due to an early `return`) when calling `set_gutter_draw(<index>, true)` from GDScript.

Instead of calling it in `TextEdit::add_gutter`, how does the following modification to `TextEdit::set_gutter_draw` look?

```
void TextEdit::set_gutter_draw(int p_gutter, bool p_draw) {
	ERR_FAIL_INDEX(p_gutter, gutters.size());
        // Instead of returning early, always call _update_gutter_width.
	if (gutters[p_gutter].draw != p_draw) {
		gutters.write[p_gutter].draw = p_draw;
	}
	_update_gutter_width();
}
```

An alternative solution is to change the default value to `false` for `GutterInfo::draw`, but I hesitate to modify it due to possible side-effects.

Which solution is preferable?

If you have another solution, please let me know.  Thanks!

Paulb23 commented:
Ah I see, in that case probably best to leave as is, only need to fix up the unit tests.

Tim-Fronsee commented:
> Ah I see, in that case probably best to leave as is, only need to fix up the unit tests.

I updated the subcase involving adding & removing gutters in the `TextEdit` gutters test case.  

Also, I modified the width checks to use a sum as I think it makes them a bit clearer.   If that's undesired, let me know and I'll change those back.

Paulb23 commented:
No strong feelings on that, I think it was just in case we wanted to make some small adjustments to padding etc without having to redo the tests.


The unit tests are failing here, probably due to the gutter width now being added.
```
./tests/scene/test_code_edit.h:3254: ERROR: CHECK( code_edit->get_text_for_symbol_lookup() == "this is s" + String::chr(0xFFFF) + "ome text" ) is NOT correct!
  values: CHECK( this is ￿some text == this is s￿ome text )
```


There's also a formatting issue would recommend setting up clang-format locally, can use the guide here: https://docs.godotengine.org/en/stable/community/contributing/code_style_guidelines.html

Tim-Fronsee commented:
> No strong feelings on that, I think it was just in case we wanted to make some small adjustments to padding etc without having to redo the tests.
> 
> The unit tests are failing here, probably due to the gutter width now being added.
> 
> ```
> ./tests/scene/test_code_edit.h:3254: ERROR: CHECK( code_edit->get_text_for_symbol_lookup() == "this is s" + String::chr(0xFFFF) + "ome text" ) is NOT correct!
>   values: CHECK( this is �some text == this is s�ome text )
> ```
> 
> There's also a formatting issue would recommend setting up clang-format locally, can use the guide here: https://docs.godotengine.org/en/stable/community/contributing/code_style_guidelines.html

Got it.  All checks passed on my fork, it seems.

Paulb23 reviewed (approved):
Thanks for the contribution!

LGTM.

akien-mga commented:
Thanks! And congrats for your first merged Godot contribution :tada:
